src/dbnode/storage/index.go

```diff
@@ -524,8 +539,15 @@ func (i *nsIndex) writeBatches(
 				batch.MarkUnmarkedEntryError(m3dberrors.ErrTooPast, idx)
 				return
 			}
+
+			if i.forwardIndexDice.forwardIndexing(entry.Timestamp) {
+				forwardWriteEntry := entry
```

Comment:
Is this a safe clone? There are three fields that are "pointer" type: 

1) OnIndexSeries
2) result
3) resultVal

Regardless I wonder if its worth adding a clone() method so that even if this operation is safe now its more likely to stay safe in the future